database.cc

```diff
@@ -1558,7 +1558,7 @@ static future<> maybe_handle_reorder(std::exception_ptr exp) {
 }
 
 future<> database::apply_with_commitlog(column_family& cf, const mutation& m, db::timeout_clock::time_point timeout) {
-    if (cf.commitlog() != nullptr) {
+    if (cf.commitlog() != nullptr && find_keyspace(cf.schema()->ks_name()).metadata()->durable_writes()) {
```

Comment:
That kind of string-based lookup is not fortunate on the fast path, but I suppose it's a de-facto standard to extract the keyspace from the schema in database.cc... no changes needed in this patch, but I wonder if it's worth it to cache the keyspace object somewhere in the schema.

Reply:
Two options:
 1. Store a pointer to the keyspace in the table
 2. Change table to have a copy of durable_writes(), and change alter_keyspace to propagate the change in durable_writes to all tables it manages.

The second would be fastest.

Reply:
Indeed, perhaps with a small comment that it's a potential reactor stall for people with too many tables, unless it's done with a Seastar loop primitive. But these people already have more sources of reactor stalls anyway.

Reply:
A loop over 5000 tables to set a flag will likely not cause a reactor stall. Even if all those flags are outside cache (likely), the CPU can issue ~10 writes in parallel, at 100ns/write that's 100M writes/sec. So 5000 writes would take 50usec, far below our stall threshold.

Reply:
I wanted to store a pointer to the keyspace in the table, but it turned out that keyspaces are stored in absl::flat_hash_map which invalidates pointers on rehash. So I put a copy of durable_writes in a shared bool (https://github.com/michoecho/scylla/commit/5544f9e4bf73d5759167ee6e24c63f8001647c95), but decided that it's ugly and went with string-based lookup for the sake of patch clarity. I guess looping over the tables is the best option here.

Reply:
Something like keyspace is better kept in a stable address, it's a time bomb if it gets moved after something captures the address.

Please do that in a separate commit with its own issue, since I'll want to backport it.

Reply:
(btw, there are other reasons to avoid having a table->keyspace pointer, it creates another dependency loop)